helpdesk-service/helpdesk-service-server/src/test/java/org/symphonyoss/symphony/bots/helpdesk/service/steps/MakerCheckerSteps.java

```diff
@@ -0,0 +1,243 @@
+
+  @Autowired
+    private TestRestTemplate restTemplate;
+
+  @Given("a system state")
+  public void doSomething() {
+  }
+
+  @When("call the create makerchecker API")
+  public void callMakercheckerAPI() {
+
+    Makerchecker makerchecker = createMakerchecker();
+
+    ResponseEntity<Makerchecker> responseEntity = restTemplate.postForEntity("/v1/makerchecker" , makerchecker, Makerchecker.class);
+
+    assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+    assertEquals(makerchecker, responseEntity.getBody());
+
+    LOGGER.info("Calling the API");
+
+  }
+
+  @When("call the create makerchecker API with invalid id")
+  public void callMakercheckerAPIWithInvalidId() {
+
+    Makerchecker makerchecker = createMakerchecker();
+    makerchecker.setId(null);
+
+    ResponseEntity<Makerchecker> responseEntity = restTemplate.postForEntity("/v1/makerchecker" , makerchecker, Makerchecker.class);
+
+    assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
+
+    LOGGER.info("Calling the API");
+
+  }
+
+  @When("call the create makerchecker API with invalid stream id")
+  public void callMakercheckerAPIWithInvalidStreamId() {
+
+    Makerchecker makerchecker = createMakerchecker();
+    makerchecker.setStreamId(null);
+
+    ResponseEntity<Makerchecker> responseEntity = restTemplate.postForEntity("/v1/makerchecker" , makerchecker, Makerchecker.class);
+
+    assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
+
+    LOGGER.info("Calling the API");
+
+  }
+
+  @When("call the create makerchecker API with invalid maker id")
+  public void callMakercheckerAPIWithInvalidMakerId() {
+
+    Makerchecker makerchecker = createMakerchecker();
+    makerchecker.makerId(null);
+
+    ResponseEntity<Makerchecker> responseEntity = restTemplate.postForEntity("/v1/makerchecker" , makerchecker, Makerchecker.class);
+
+    assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
+
+    LOGGER.info("Calling the API");
+
+  }
+
+  @When("call the create makerchecker API with invalid state")
+  public void callMakercheckerAPIWithInvalidState() {
+
+    Makerchecker makerchecker = createMakerchecker();
+    makerchecker.setId("NEW_MOCK_MAKERCHECKER_ID");
+    makerchecker.setState(MakercheckerClient.AttachmentStateType.DENIED.getState());
+
+    ResponseEntity<Makerchecker> responseEntity = restTemplate.postForEntity("/v1/makerchecker" , makerchecker, Makerchecker.class);
+
+    assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, responseEntity.getStatusCode());
+
+    LOGGER.info("Calling the API");
+
+  }
+
+  @When("call the create makerchecker API with same id")
+  public void callMakercheckerAPIWithSameId() {
+
+    Makerchecker makerchecker = createMakerchecker();
+
+    ResponseEntity<Makerchecker> responseEntity = restTemplate.postForEntity("/v1/makerchecker" , makerchecker, Makerchecker.class);
+
+    assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, responseEntity.getStatusCode());
```

Comment:
Should it return HTTP 500?

Reply:
When we try to insert in the DB the same id, it returns the HTTP 500 but I don't know if this is correct.. it should be HTTP 400? Bad Resquest.